weather_dl/download_pipeline/manifest.py

```diff
@@ -242,22 +447,138 @@ def _update(self, download_status: DownloadStatus) -> None:
                 logger.debug(download_status)
 
 
+class BQManifest(Manifest):
+    """Writes a JSON representation of the manifest to BQ file.
+
+    This is an append-only implementation, the latest value in the manifest
+    represents the current state of a download.
+    """
+    def __init__(self, location: Location) -> None:
+        super().__init__(Location(location[5:]))
+        TABLE_SCHEMA = [
+            bigquery.SchemaField('config_name', 'STRING', mode='REQUIRED',
+                                 description="The name of the config file associated with the request."),
+            bigquery.SchemaField('selection', 'JSON', mode='REQUIRED',
```

Comment:
Hey Rahul. I'm talking to Tyler about his dashboard (that doesn't use this DB... yet). He needs some way to calculate the total number of shards in order to display the percentage complete. Any ideas to accomplish this?

Reply:
Yeah we can do that. Basically in partition step as soon as the shards are created it will update the manifest stating it is `scheduled` (i.e. status: scheduled). This allows us to obtain the total number of shards for a particular config.
To determine the number of shards that have been successfully downloaded, we can check for shards with a stage of 'upload' and a status of 'success'.

Reply:
Brilliant. I like this clean, simple solution.